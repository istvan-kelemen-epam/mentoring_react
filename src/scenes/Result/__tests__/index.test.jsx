import { Result } from '../index.jsx';
import renderer from 'react-test-renderer';

describe('scenes/Result', () => {
	let instance;

	beforeEach(() => {
		instance = Object.create(Result.prototype);
	});

	describe('.render', () => {
		it('should create empty result', () => {
			instance.props = {};
			const rendered = instance.render();
			const snapshot = renderer.create(rendered);
			expect(snapshot).toMatchSnapshot();
		});

		it('should create "not found" result', () => {
			instance.props = {data: []};
			const rendered = instance.render();
			const snapshot = renderer.create(rendered);
			expect(snapshot).toMatchSnapshot();
		});

		it('should create list result', () => {
			instance.props = {data: [{
				id: 119556,
				title: 'Kill Bill',
				year: 2003,
				genre: 'Action & Adventure'
			}]};
			const rendered = instance.render();
			const snapshot = renderer.create(rendered);
			expect(snapshot).toMatchSnapshot();
		});
	});
});