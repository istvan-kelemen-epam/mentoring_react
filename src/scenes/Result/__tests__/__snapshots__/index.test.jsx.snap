// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scenes/Result .render should create "not found" result 1`] = `
<section
  className="result"
>
  <div
    className="result__not-found"
  >
    No films found
  </div>
</section>
`;

exports[`scenes/Result .render should create empty result 1`] = `
<section
  className="result"
/>
`;

exports[`scenes/Result .render should create list result 1`] = `
<section
  className="result"
>
  <ul
    className="result-list"
  >
    <li
      className="result-list__item"
    >
      <div
        className="result-list__item-title"
      >
        Kill Bill
      </div>
      <div
        className="result-list__item-year"
      >
        2003
      </div>
      <div
        className="result-list__item-genre"
      >
        Action & Adventure
      </div>
    </li>
  </ul>
</section>
`;
